### DISCUSSION TITLE
Doubt about the use of GPR sgy format for TT

### DISCUSSION THREAD

**Discussion by bralec** [gimli-org/pyGIMLi#473] [Ideas]
### Problem description

Hello, I hope everything is going very well.

I have some GPR data in sgy format, and I have been trying to be able to use it in the TravelTime tool. The issue is that I have not been able to find something similar to the unified data format used in ERT, or a direct reading of it. I would like to know then, is there any way to be able to use this GPR data format in pyGIMLi?

Thank you in advance for your help.

**Comment by halbmy** [comment]
You're thinking of first-arrivals and not full waveforms, right?

There are several SEGY readers around, so if you have an example it would probably not a big deal to enable such a reader once the format (SEGY is rather flexible) is defined.

**Comment by bralec** [comment]
If the first-arrival is like an inversion, then yes.

I've had a second look at some of the ones you mention, but I'm more interested in studying the velocity you get. Also, with that I could work more closely with the ERT and GPR data, and since you could somehow, that would be great.

**Comment by halbmy** [comment]
Travel-time tomography is inversion, and that's what pyGIMLi does, i.e. generating a velocity model from the first-arrival traveltimes. So if you put your traveltimes into a SEGY file along with all the geometry needed, we could provide a solution if this is a typical workflow that also helps others.

**Comment by bralec** [comment]
I understand, thank you very much for the clarification. 

In that case I will look into the structure of my data to see if it meets the parameters, and for the time being I will share a copy of the file in case it can serve as support.

[Test.zip](https://github.com/gimli-org/gimli/files/8602541/Test.zip)

**Comment by bralec** [comment]
Hello again @halbmy, I hope everything is going well.

Regarding everything you had told me, I was able to solve several issues:

**First**
Thanks to the **segysak** library, which imports the data into a **xarray** type dataset (based on dataframe), I was able to extract the information that is measured with our GPR equipment.

```python
from segysak.segy import segy_loader

import matplotlib.pyplot as plt
import numpy as np

fileName = "DAT_0096_A1.sgy"
dataset = segy_loader(fileName)
print(dataset)
```

In this case, our GPR equipment is set to antenna-dependent signal speed (in Hz), which results in data on common depth points (CDP) and their respective two-way time (TwT).
```python
CDP = np.array(dataset.data.cdp)
TwT = np.array(dataset.data.twt)
```
In addition, we can extract other important data such as

- dataset.sample_rate
- dataset.percentiles
- dataset.coord_scalar

or plot a preview with the data
```python
ax = dataset.data.transpose("twt", "cdp").plot()
plt.xlim(min(CDP), max(CDP))
plt.xlabel("Common Depth Point (CDP)")
plt.ylim(25, 65)
plt.ylabel("Two-way Time (TWT)")
```
![preview](https://user-images.githubusercontent.com/82900508/183980187-206cc8b1-67fc-477f-9c53-8b62e9dd0897.png)

**Second** 
I don't know if this example that you asked me for, and that I am giving you, is enough because the question arises as to whether the TwT is the first-arrivals or the full waveforms, of course. If I had to give an answer, from my perspective, it would be that it is the complete waveforms, because it is the times to and from the transmitted signal (Tx).

Otherwise, if it is the traveltimes of the first-arrival and meets the geometry required, that would be great.

>Command for segysek installation: pip install segysek
>File: [DAT_0096_A1.zip](https://github.com/gimli-org/gimli/files/9302808/DAT_0096_A1.zip)

**Comment by GGDRriedel** [comment]
What do you want to do with the data. Because it seems like there are some missconceptions here.
You are mixing up seismic CDPs with GPR cdps. 

Unless you have a funky Antenna, your GPR cdp will always be the midpoint beween Rx and Tx(receiver and transmitter), aka the middle of your antenna. 
Your TWT is simply the recording time of your signal. 

It also helps to do np.flipud your data for more reality-based plotting. 

What do you want to invert here? It seems like 400 MHz data.

**Comment by bralec** [comment]
Hello @GGDRriedel. Thank you very much for the explanation and clarification of the data. It is suitable for me to clarify these issues.

In the first instance, I want to be able to do the travel-time tomography in pyGIMLi. The initial problem was to be able to extract the data from the SGY format, but now it would be to see if there is a way to use it for inversion.

PD: I will test the np.flipup in the preview from the data.

**Comment by GGDRriedel** [comment]
I honsestly don't get how you want to do a traveltime tomography with GPR zero-offset data.

**Comment by bralec** [comment]
Would this be the data for precise zero-time adjustment? Because for this GPR profile they are not the only data I have.
